lib/types/uuid.js

```diff
@@ -71,7 +73,7 @@ class Uuid {
      * @param {Uuid} other The other value to test for equality.
      */
     equals(other) {
-        return other instanceof Uuid && this.buffer.equals(other.buffer);
+        return other instanceof Uuid && this.toString() === other.toString();
```

Comment:
I'm strongly against stringification for comparison purposes, especially that this allocates. Can't we just compare these types byte by byte?